types/got/index.d.ts

```diff
@@ -198,6 +199,8 @@ declare namespace got {
         hooks?: Hooks<GotFormOptions<E>, Record<string, any>>;
     }
 
+    type RequestFunction = typeof https.request;
```

Comment:
`typeof https.request` is `https.ClientRequest`, but there is [note](https://github.com/sindresorhus/got/tree/v9.6.0#request-1) in documentation. 

`request` property is just an object, which contains `gotOptions` property: https://github.com/sindresorhus/got/blob/a45e071e8dc5027ab9c7fad5919195501cd9e9ca/source/request-as-event-emitter.js#L108

Reply:
I'm sorry. This PR about `options.request`, not `response.request`.